lib/aba.rb

```diff
@@ -5,40 +5,95 @@
 class Aba
   include Aba::Validations
 
-  attr_accessor :bsb, :financial_institution, :user_name, :user_id, :description, :process_at
+  attr_accessor :bsb, :financial_institution, :user_name, :user_id, :description, :process_at, :transactions
 
-  validates_bsb :bsb, allow_blank: true
+  # BSB
+  validates_bsb         :bsb, allow_blank: true
 
-  validates_max_length :user_name,              26
-  validates_max_length :user_id,                6
-  validates_max_length :description,            12
+  # Financial Institution
+  validates_length      :financial_institution, 3
 
-  validates_length     :financial_institution,  3
-  validates_length     :process_at,             6
+  # User Name
+  validates_presence_of :user_name
+  validates_max_length  :user_name, 26
+  validates_becs        :user_name
 
-  def initialize(attrs = {})
+  # User ID
+  validates_presence_of :user_id
+  validates_max_length  :user_id, 6
+  validates_integer     :user_id, false
+
+  # Description
+  validates_max_length  :description, 12
+  validates_becs        :description
+
+  # Process at Date
+  validates_length      :process_at, 6
+  validates_integer     :process_at, false
+
+
+  def initialize(attrs = {}, transactions = [])
     attrs.each do |key, value|
       send("#{key}=", value)
     end
 
-    @transactions = []
+    @transaction_index = 0
+    @transactions = {}
+
+    unless transactions.nil? || transactions.empty?
+      transactions.to_a.each do |t|
+        self.add_transaction(t) unless t.nil? || t.empty?
+      end
+    end
 
     yield self if block_given?
   end
 
   def to_s
```

Comment:
Probably we can leave it as it is. I'm used to the rails style where bangs can indicate that a method can throw an exception.